internal/dumper/loader.go

```diff
@@ -62,37 +76,59 @@ func (l *Loader) Run(ctx context.Context) error {
 	pool.Put(conn)
 
 	// tables.
-	conn = pool.Get()
-	if err := l.restoreTableSchema(l.cfg.OverwriteTables, files.schemas, conn); err != nil {
-		return err
+	if l.canRestoreSchema() {
+		conn = pool.Get()
+		if err := l.restoreTableSchema(l.cfg.OverwriteTables, files.schemas, conn); err != nil {
+			return err
+		}
+		pool.Put(conn)
+	} else {
+		l.cfg.Printer.Println("Skipping restoring schema files...")
 	}
-	pool.Put(conn)
 
+	// Commenting out to add some predictability to the order in which data files will be processed:
 	// Shuffle the tables
-	for i := range files.tables {
-		j := rand.Intn(i + 1)
-		files.tables[i], files.tables[j] = files.tables[j], files.tables[i]
-	}
+	//for i := range files.tables {
+	//	j := rand.Intn(i + 1)
+	//	files.tables[i], files.tables[j] = files.tables[j], files.tables[i]
+	//}
```

Comment:
I don't know if MySQL gives any guarantees about table ordering? So the start / end table logic is maybe not something very reliable to add. 

Otherwise we might need to change the query to show the tables so we can enforce the order there? That means changing it to use the `information_schema` tables then. 

In any way, if we remove code, we should remove it, not comment it :smile:. We have `git` to track things and find old code :wink:.

Reply:
I appreciate the input @dbussink ❤️!

Is your first sentence related to the `--starting-table` and `--ending-table` options? 

Those primarily help to drive the list of files that are being picked up from within the dump folder and mainly help provide some extra control over which files are going to be selected and used for the current `restore-dump` operation so the MySQL table ordering shouldn't necessarily factor in with those options?

I mainly see the `--starting-table`, `--ending-table` and several of the other new options added with this PR help ing to avoid needing to create multiple copies of a dump folder since normally you might have to create a new copy of an existing dump folder and then perform operations on it to keep just the tables / files you want to restore if you want to restore less than the full dataset.
---
As for the shuffling related code I commented out, the purpose of that code was initially confusing for me since I thought it did something different and I was unsure what the original rationale for adding it was.

I can definitely remove it completely from the file though and will make that correction shortly!

The shuffling only affects the order in which the data files get processed.

For example, with shuffling the data files will get processed out of order:

```
Started Processing Data File: metrics.connections.00005.sql in thread 3 (File 1 of 5)
Started Processing Data File: metrics.connections.00004.sql in thread 4 (File 2 of 5)
Started Processing Data File: metrics.connections.00001.sql in thread 1 (File 3 of 5)
Started Processing Data File: metrics.connections.00002.sql in thread 2 (File 4 of 5)
```

I only have 5 files in this example so it's not a great one but you can imagine things possibly starting at file 500, 1, 10, and 42 for a table with the shuffling option in place and then on a subsequent execution the data files would be processed again in a different order, etc.

I think removing the shuffling code is better for users since knowing that the files will be processed more or less in numeric order seems like a good thing? There is the caveat of threading still allowing for some processing out of order, but at least you can see that files 1-4 were started first below and then it will progress to file 5 afterward:

```
Started Processing Data File: metrics.connections.00001.sql in thread 3 (File 1 of 5)
Started Processing Data File: metrics.connections.00004.sql in thread 2 (File 4 of 5)
Started Processing Data File: metrics.connections.00003.sql in thread 1 (File 3 of 5)
Started Processing Data File: metrics.connections.00002.sql in thread 4 (File 2 of 5)
```

This seems like it would be useful for users since they would have some predictability for the order in which their data files will be processed by the `restore-dump` operation. 

This also comes out of working with a current user that is migrating from Postgres and generating files in the format needed for `restore-dump`.

Throughout that process different SQL errors have cropped up and it can be unclear which file needs to be reviewed more closely without the extra output that `--show-details` now provides. 

With the extra context of where you are in the processing of the files with `--show-details` this can help point a user in the right direction more quickly and they may even be able to more easily know where they can try and continue from in a subsequent `restore-dump` attempt without necessarily having to restore all of their data files again.

Going back to the shuffling example once more, if data file 500 for a table contained a problematic row when running a `restore-dump` attempt the amount of time it would take to encounter the error right now would vary since it would depend on where that file lands in the shuffled ordering. Without the shuffling we would find that the error should be encountered a fairly consistent manner since the data files will be processed in the same order.

Reply:
I removed the commented out shuffling code.